src/modules/jcms_notifications/src/NotificationService.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Drupal\jcms_notifications;
+
+use Aws\AwsClientInterface;
+use Aws\Sns\SnsClient;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Class NotificationService.
+ *
+ * @package Drupal\jcms_notifications
+ */
+class NotificationService {
+
+  /**
+   * @var \Aws\Sns\SnsClient
+   */
+  protected $snsClient;
+
+  protected $endpoint = '';
+
+  protected $topicName = '';
+
+  protected $region = '';
+
+  /**
+   * NotificationService constructor.
+   *
+   * @param \Aws\AwsClientInterface|NULL $sns_client
+   */
+  public function __construct(AwsClientInterface $sns_client = NULL) {
+    $this->endpoint = Settings::get('jcms_sqs_endpoint');
+    $this->region = Settings::get('jcms_sqs_region');
+    $this->topicName = Settings::get('jcms_sns_topic');
+    $config = [
+      'profile' => 'default',
+      'version' => 'latest',
+      'region' => $this->region,
+    ];
+    if (!empty($this->endpoint)) {
+      $config['endpoint'] = $this->endpoint;
+    }
+    $this->snsClient = $sns_client ?: new SnsClient($config);
+  }
+
+  /**
+   * Sends a notification message to SNS.
+   *
+   * @param array $message
+   *
+   * @return \Aws\Result
+   */
+  public function sendNotification(array $message) {
+    return $this->snsClient->publish([
+      //'TopicArn' => $topic['TopicArn'],
+      'TopicArn' => $this->topicName,
```

Comment:
the topic ARN (Amazon Resource Name?) and the topic name are not the same thing: the name is like `bus-blog-articles--end2end` while the ARN is `arn:aws:sns:us-east-1:512686554592:bus-blog-articles--end2end`. You can probably look up the ARN using the SDK like this:
https://github.com/elifesciences/bus/blob/master/send-example.php#L20
but it can be a mess as a wrong topic name will just create it. So, expanding on the previous comment, what about we configure this value from the formula:
```
arn:aws:sns:us-east-1:512686554592:bus-%s--end2end
```
so that you just have to substitute the placeholder with the content type? The strange number is just the AWS account id, nothing fancy.